app/src/marketplace/utils/web3.js

```diff
@@ -58,32 +63,78 @@ export const getBalances = (): Promise<[BN, BN, BN]> => {
 }
 
 /**
- * Returns the DAI and ETH value equivalents for a given quanity of DATA through a 'Uniswap'.
- * A safety margin is added to ensure transactions succeed incase of slippage and token volatility.
+ * uniswapDATAtoETH & uniswapDATAtoDAI the DAI and ETH value equivalents for a given quanity of DATA through a 'Uniswap'.
+ * A safety margin is added to ensure transactions succeed due to dynamic mainnet DATA exchange rates.
+ *
+ * !!!Note some versions of web3 will return a BigNumber from the Ethers library. This has a
+ * different prototype compared with BigNumber (BN). This is why extra BNs are used in this function.
  * @param dataQuantity Number of DATA coins.
- */
-export const getUniswapEquivalents = async (dataQuantity: string): Promise<[BN, BN]> => {
-    const web3 = getWeb3()
-    const amount = web3.utils.toWei(dataQuantity)
-    const DATA = process.env.DATA_TOKEN_CONTRACT_ADDRESS
-    const ETH = '0x0000000000000000000000000000000000000000'
-    const DAI = process.env.DAI_TOKEN_CONTRACT_ADDRESS
-    const safetyMargin = 1.05
-
-    let ethValue = await call(uniswapAdaptorMethods().getConversionRate(DATA, ETH, amount))
-    ethValue = BN(ethValue).toNumber() * safetyMargin
-    ethValue = BN(web3.utils.fromWei(ethValue.toString()))
-
-    let daiValue = await call(uniswapAdaptorMethods().getConversionRate(DATA, DAI, amount))
-    daiValue = BN(daiValue).toNumber() * safetyMargin
-    daiValue = BN(web3.utils.fromWei(daiValue.toString()))
-
-    return [ethValue, daiValue]
+*/
+export const uniswapDATAtoETH = async (dataQuantity: string, usePublicNode: boolean = false): Promise<BN> => {
+    if (dataQuantity !== '0') {
```

Comment:
What's the benefit of returning infinity instead of throwin an error?

Reply:
I think both ways could be implemented/handled without much of a quantifiable difference. I chose returning an invalid number, rather than handling an error because it fits into the validation logic inside the ChooseAccessPeriodDialog quite well. 

Handling an error seems a touch heavy handed because, for example, a high purchase price will break the contract call required for a USD approximation of ETH, but not the actual DATA -> ETH conversion contract call (probably depending on the pool size of one or the other). If the approximation fails, we don't care so much.

Reply:
Ok, makes sense... maybe add that notice to the comment also.

Reply:
Good point. Done.